synedrion/src/cggmp21/sigma/aff_g.rs

```diff
@@ -46,15 +46,16 @@ pub(crate) struct AffGPublicInputs<'a, P: SchemeParams> {
     /// Paillier ciphertext $Y = enc_1(y, \rho_y)$.
     pub cap_y: &'a Ciphertext<P::Paillier>,
     /// Point $X = g * x$, where $g$ is the curve generator.
-    pub cap_x: &'a Point,
+    pub cap_x: &'a Point<P>,
 }
 
 /// ZK proof: Paillier Affine Operation with Group Commitment in Range.
 #[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(bound(deserialize = "for<'x> P: Deserialize<'x>"))]
```

Comment:
It's kind of weird to see these bounds everywhere. I thought about dealing with #10 by bounding `SchemeParams` on `Serialize`/`Deserialize`, which you did, but I would expect that to eliminate the need for the explicit bounds?

Reply:
This stuff was driving me nuts. I cannot tell you why this works or if there are other/better ways of doing it. It was by far the most frustrating thing about this work. You add a bound in one place and it typechecks, but then breaks somewhere else, you fix it over there in a way that makes sense and now it breaks somewhere else...

Reply:
Filed https://github.com/serde-rs/serde/issues/2899. These bounds can be avoided if we don't bound `SchemeParams` on `Deserialize`, but then other explicit bounds will be needed - but we already had those, so maybe that's the best course of action.

Reply:
Did this in [https://github.com/entropyxyz/synedrion/pull/186/commits/7a1a5745e65fc7f7c78308a779450b26aaccd1d9] and it's probably a bit better than having a `Deserialize` on `SchemeParams` – at least it's a bit more explicit. 
That said it's both noisy and weird: sometimes explicit bounds are required and [sometimes not](https://github.com/entropyxyz/synedrion/pull/186/commits/7a1a5745e65fc7f7c78308a779450b26aaccd1d9#diff-f55b0977d14a290d03f0fcbded011e7fceb653ba15d2b4693d059183405de423R61), and I don't quite understand why.